packages/scandipwa/src/component/RenderWhenVisible/RenderWhenVisible.component.js

```diff
@@ -52,16 +51,31 @@ export class RenderWhenVisible extends PureComponent {
         }
     };
 
-    shouldRender(isVisible) {
-        return isVisible || this.wasVisible;
+    shouldRender() {
+        const { wasVisible } = this.state;
+        return !wasVisible && !this.isSSR() && !this.isCrawler();
     }
 
     handleVisibilityToggle = (isVisible) => {
-        if (!this.wasVisible && isVisible) {
+        const { wasVisible } = this.state;
+
+        if (!wasVisible && isVisible) {
             this.setState({ wasVisible: true });
         }
     };
 
+    isSSR() {
+        return !globalThis.window || !globalThis.window.document;
+    }
+
+    isCrawler() {
```

Comment:
1. Consider making this an util. It is not strictly related to this component. We may need to determine whether user is crawler somewhere else in the future :)
2. Please have this regex declared separately, outside the function scope, there is no need to re-declare it each time this function is called.

Reply:
Done